common/persistence/nosql/nosqlplugin/interfaces.go

```diff
@@ -509,3 +703,20 @@ const (
 	SortByStartTime VisibilitySortType = iota
 	SortByClosedTime
 )
+
+const (
+	CurrentWorkflowWriteModeNoop CurrentWorkflowWriteMode = iota
+	CurrentWorkflowWriteModeUpdate
+	CurrentWorkflowWriteModeInsert
+)
+
+func (w *CurrentWorkflowWriteCondition) GetCurrentRunID() string {
+	if w == nil || w.CurrentRunID == nil {
+		return ""
+	}
+	return *w.CurrentRunID
+}
+
+func (e *WorkflowOperationConditionFailure) Error() string {
```

Comment:
Do we need to update the impl for `IsConditionFailedError()` to also check for this error?

Reply:
No we don't need to. Also checking that means you have double checking, which is duplicated and unnescessary. 

I will remove IsConditionFailedError() from the interface in another PR once this one is merged: https://github.com/uber/cadence/issues/4273

Reply:
you are right. I found the corresponding error handling logic for the new type.